samples/EverythingServer/LoggingUpdateMessageSender.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using ModelContextProtocol.Protocol.Messages;
+using ModelContextProtocol.Protocol.Types;
+using ModelContextProtocol.Server;
+
+namespace EverythingServer;
+
+public class LoggingUpdateMessageSender(IMcpServer server) : IHostedService
+{
+    readonly Dictionary<LoggingLevel, string> _loggingLevelMap = new()
+    {
+        { LoggingLevel.Debug, "Debug-level message" },
+        { LoggingLevel.Info, "Info-level message" },
+        { LoggingLevel.Notice, "Notice-level message" },
+        { LoggingLevel.Warning, "Warning-level message" },
+        { LoggingLevel.Error, "Error-level message" },
+        { LoggingLevel.Critical, "Critical-level message" },
+        { LoggingLevel.Alert, "Alert-level message" },
+        { LoggingLevel.Emergency, "Emergency-level message" }
+    };
+
+    public async Task StartAsync(CancellationToken cancellationToken)
+    {
+        var currentLevel = server.Services!.GetRequiredService<Func<LoggingLevel>>();
+
+        while (!cancellationToken.IsCancellationRequested)
```

Comment:
This won't get canceled during host shutdown since it's only for StartAsync. You should probably implement BackgroundService.ExecuteAsync instead of IHostedService directly similar to the way StdioMcpServerHostedService does.